src/py/crankshaft/crankshaft/analysis_data_provider.py

```diff
@@ -48,20 +52,22 @@ def get_nonspatial_kmeans(self, query):
         """fetch data for non-spatial kmeans"""
         try:
             data = plpy.execute(query)
+            verify_data(len(data))
             return data
         except plpy.SPIError, err:
             plpy.error('Analysis failed: %s' % err)
 
     def get_spatial_kmeans(self, params):
         """fetch data for spatial kmeans"""
         query = ("SELECT "
-                 "array_agg({id_col} ORDER BY {id_col}) as ids,"
-                 "array_agg(ST_X({geom_col}) ORDER BY {id_col}) As xs,"
-                 "array_agg(ST_Y({geom_col}) ORDER BY {id_col}) As ys "
+                 "array_agg(\"{id_col}\" ORDER BY \"{id_col}\") as ids,"
+                 "array_agg(ST_X(\"{geom_col}\") ORDER BY \"{id_col}\") As xs,"
+                 "array_agg(ST_Y(\"{geom_col}\") ORDER BY \"{id_col}\") As ys "
                  "FROM ({subquery}) As a "
-                 "WHERE {geom_col} IS NOT NULL").format(**params)
+                 "WHERE \"{geom_col}\" IS NOT NULL").format(**params)
```

Comment:
I've found that these statements can be written more readably by using triple quotes:

```
'''
SELECT "{your_fancy_column}"
FROM "{your_fancy_table}"
WHERE "{other_column}" = '{val}'
'''.format(**params)
```